lib/playlistGenerator/PlaylistGenerator.js

```diff
@@ -233,9 +233,9 @@ var doPersist =  function() {
     that.emit('beforePlaylistChanged', that);
 
     that.isDirty = false;
-    if(that.playlistImp.inner.liveWindowDuration) {
-        that.playlistImp.inner.presentationEndTime = that.playlistImp.getCurrentTime() + that.playlistImp.inner.liveWindowDuration;
-    }
+    that.playlistImp.inner.presentationEndTime = that.playlistImp.getCurrentTime() + that.playlistImp.inner.liveWindowDuration;
```

Comment:
1. i think the logic should be in one place, both for this place and the  CrossDCsStreamPersistence
right now it's duplicated 
2. 
if one of the DC is down then
we want the packager to return 404 after X seconds  (in your case it will be the DVR window)

if both DC are down then:
we want to return END-LIST for the DVR window  (now +  liveWindowDuration)